addons/sourcemod/scripting/ConnectAnnounce.sp

```diff
@@ -512,38 +589,34 @@ stock Action SQLSelect_Join(Handle timer, any data)
 	char sQuery[MAX_SQL_QUERY_LENGTH];
 	char sClientSteamID[32];
 
-	pack.ReadCell();
+	int client = pack.ReadCell();
 	pack.ReadString(sClientSteamID, sizeof(sClientSteamID));
 
-	Format(sQuery, sizeof(sQuery), "SELECT `message` FROM `join` WHERE `steamid` = '%s';", sClientSteamID);
-	if (g_hDatabase != null)
-		SQL_TQuery(g_hDatabase, OnSQLSelect_Join, sQuery, data, DBPrio_Low);
-	else
-		SetFailState("Database error while checking \"message\" from \"join\" table.");
-
-	return Plugin_Stop;
-}
-
-public void OnSQLSelect_Join(Handle hParent, Handle hChild, const char[] err, any data)
-{
-	DataPack pack = view_as<DataPack>(data);
-	pack.Reset();
-
-	int client = pack.ReadCell();
+	delete pack;
 
-	if (hChild == null)
-	{
-		LogError("An error occurred while querying the database for the join message. (%s)", err);
-		return;
-	}
-	else if (SQL_FetchRow(hChild))
+	if (DB_Connect())
 	{
-		SQL_FetchString(hChild, 0, g_sClientJoinMessage[client], sizeof(g_sClientJoinMessage[]));
+		Format(sQuery, sizeof(sQuery), "SELECT `message` FROM `join` WHERE `steamid` = '%s';", sClientSteamID);
+
+		DBResultSet result = SQL_Query(g_hDatabase, sQuery);
```

Comment:
g_hDatabase.Query to keep every database query threaded and not wait on the main thread